modules/rounds.js

```diff
@@ -226,6 +227,23 @@ Rounds.prototype.tick = function (block, done) {
 					}
 				});
 			}
+		}).then(function () {
+			// Check if we are one block before last block of round, if yes - perform round snapshot
+			if ((block.height+1) % slots.delegates === 0) {
+				library.logger.debug('Performing round snapshot...');
+
+				return t.batch([
+					t.rounds.clearRoundSnapshot(),
+					t.rounds.performRoundSnapshot(),
+					t.rounds.clearVotesSnapshot(),
+					t.rounds.performVotesSnapshot()
+				]).then(function () {
+					library.logger.trace('Round snapshot done');
+				}).catch(function (err) {
+					library.logger.error('Round snapshot failed', err);
+					return Promise.reject(err);
```

Comment:
Recommended standard in this case is to do `throw err;` instead ;)

Reply:
Tahnks, fixed. :)